module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -1821,6 +1781,151 @@ protected function getItemStatusCodes($item)
     }
 
+    /**
+     * Get item status code for CheckedOut status
+     *
+     * @param string $code Status code
+     * @param array  $data Status data
+     * @param array  $item Item
+     *
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function getStatusCodeItemCheckedOut($code, $data, $item)
+    {
+        $overdue = false;
+        if (!empty($data['due_date'])) {
+            $duedate = $this->dateConverter->convert(
+                'Y-m-d',
+                'U',
+                $data['due_date']
+            );
+            $overdue = $duedate < time();
+        }
+        return $overdue ? 'Overdue' : 'Charged';
+    }
+
+    /**
+     * Get item status code for Lost status
+     *
+     * @param string $code Status code
+     * @param array  $data Status data
+     * @param array  $item Item
+     *
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function getStatusCodeItemLost($code, $data, $item)
+    {
+        return 'Lost--Library Applied';
+    }
+
+    /**
+     * Get item status code for NotForLoan status
+     *
+     * @param string $code Status code
+     * @param array  $data Status data
+     * @param array  $item Item
+     *
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function getStatusCodeItemNotForLoan($code, $data, $item)
+    {
+        // NotForLoan is special: status has a library-specific
+        // status number. Allow mapping of different status numbers
+        // separately (e.g. Item::NotForLoan with status number 4
+        // is mapped with key Item::NotForLoan4):
+        $statusKey = $code . ($data['status'] ?? '-');
+        // Replace ':' in status key if used as status since ':' is
+        // the namespace separator in translatable strings:
+        return $this->itemStatusMappings[$statusKey]
+            ?? $data['code'] ?? str_replace(':', '_', $statusKey);
+    }
+
+    /**
+     * Get item status code for NorForLoanForcing status
+     *
+     * @param string $code Status code
+     * @param array  $data Status data
+     * @param array  $item Item
+     *
+     * @return string
+     */
+    protected function getStatusCodeItemNotForLoanForcing($code, $data, $item)
+    {
+        return $this->getStatusCodeItemNotForLoan($code, $data, $item);
+    }
+
+    /**
+     * Get item status code for Transfer status
+     *
+     * @param string $code Status code
+     * @param array  $data Status data
+     * @param array  $item Item
+     *
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function getStatusCodeItemTransfer($code, $data, $item)
+    {
+        $onHold = array_key_exists(
+            'Item::Held',
+            $item['availability']['notes'] ?? []
+        );
+        return $onHold ? 'In Transit On Hold' : 'In Transit';
```

Comment:
Is `'In Transit On Hold'` a typo for `'On Hold'`?

Reply:
No, it's a string that, I believe, has its roots in Voyager. It's used when the item is on hold but still in transit to the pickup location. I'm open to using something else as the default string, though I recall that libraries had some opinions on the wording including that they don't want it to convey any idea of when the item might be available for pickup.

Reply:
Okay, great, in that case I'm fine with leaving it in place. If people don't like it, they can always translate it.